File: templates/ApplicationTests.java
package ${packageName};

import org.junit.Test;
import org.junit.runner.RunWith;
${testImports}import org.springframework.boot.test.SpringApplicationConfiguration;
import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;

@RunWith(SpringJUnit4ClassRunner.class)
@SpringApplicationConfiguration(classes = Application.class)
${testAnnotations}public class ApplicationTests {

	@Test
	public void contextLoads() {
	}

}

File: initializr-generator/src/main/resources/templates/ApplicationTests.scala
package {{packageName}};

import org.junit.Test;
import org.junit.runner.RunWith;
{{testImports}}
{{#newTestInfrastructure}}
@RunWith(classOf[SpringRunner])
@SpringBootTest
{{/newTestInfrastructure}}
{{^newTestInfrastructure}}
@RunWith(classOf[SpringJUnit4ClassRunner])
@SpringApplicationConfiguration(classes = Array(classOf[{{applicationName}}]))
{{/newTestInfrastructure}}
{{testAnnotations}}class {{applicationName}}Tests {

	@Test
	def contextLoads(): Unit = {
	}

}
